src/js/controller/controller.js

```diff
@@ -918,6 +918,19 @@ Object.assign(Controller.prototype, {
             return _setItem(_model.get('item'));
         };
 
+        this.setPlaylistItem = function (index, item) {
+            item = normalizePlaylistItem(_model, item, {});
```

Comment:
Make sure to pass in existing `item.feedData` if present:

`item = normalizePlaylistItem(_model, item, item.feedData || {});`